src/server/gameEngine/gameEngine.ts

```diff
@@ -0,0 +1,125 @@
+import {
+	assignRandomColorsForPlayers,
+	createInitialGame,
+	createPlayer,
+	validateCanJoinGame,
+	validateGameExists,
+	validateGameIsInSpecificState,
+	validateUserInGame,
+} from './gameEngine.lib'
+import { getGameFromStore, getGameStore } from './store/store'
+import { EngineChessGame, EngineChessGameStatus } from './store/store.models'
+import { playMove as playChessMove } from '../../chess/moves/moves'
+import { fenToState, stateToFen } from '../../chess/serialization/fen/fen'
+
+export function getGame(gameId: string) {
+	return getGameFromStore(gameId)
+}
+
+export function createGame(playerId: string): EngineChessGame {
+	const store = getGameStore()
+	const game = createInitialGame(playerId)
+	store.set(game.id, game)
+	return game
+}
+export function joinGame(playerId: string, gameId: string): EngineChessGame {
```

Comment:
Mainly because it was the easiest way to go. GraphQL layer might want any of the object fields, so it is easy to just give the whole object for it to pick the fields from.

One other reason is because I am a believer of not mutating anything, and instead of creating changed copies while keeping the old value the same.

That is not being used here though, as mutating a single object over and over again is so much more performant (and in this case less prone to race-conditions) than creating new copies.

It is definitely not the cleanest currently, and I am looking for something better, but it was fast to implement :D

Reply:
What about

```javascript
	joinGame(
			_root: any,
			args: { gameId: string },
			ctx: GraphqlRequestContext,
		): EngineChessGame {
			const game = getGameFromStore(gameId)
                        joinGame(ctx.sessionId, game)
			publishGameStateChange(game.id, game)
			return game
		},
```
or even
```javascript
			const game = getGameFromStore(gameId)
                        game.join(ctx.sessionId)
```
?